doc/stp/SONiC_PVST_HLD.md

```diff
@@ -0,0 +1,700 @@
+    hello_time            = 2*DIGIT			            ; hello time in secs (1 to 10 sec, DEF:2sec)
+    forward_delay         = 2*DIGIT			            ; forward delay in secs (4 to 30 sec, DEF:15 sec)
+    hold_time             = 1*DIGIT			            ; hold time in secs (1 sec) 
+    last_topology_change  = 1*10DIGIT			        ; last time stamp of when topology occured TODO
+    topology_change_count = 1*10DIGIT			        ; Number of times topology change occured
+    root_bridge_id        = 16HEXDIG			        ; root bridge id
+    root_path_cost        = 1*9DIGIT			        ; port path cost 
+    desig_bridge_id       = 16HEXDIG			        ; designated bridge id
+    root_port             = ifName			                ; Root port name
+    root_max_age          = 1*2DIGIT			        ; Max age as per root bridge
+    root_hello_time       = 1*2DIGIT			        ; hello time as per root bridge
+    root_forward_delay    = 1*2DIGIT			        ; forward delay as per root bridge
+    stp_instance          = 1*4DIGIT			        ; STP instance for this VLAN
+
+### STP_VLAN_INTF_TABLE
+    ;Stores STP VLAN interface details 
+    ;Status: work in progress
+    key                 = STP_VLAN_INTF:"Vlan"vlanid:ifname     ; VLAN+Intf with prefix "STP_VLAN_INTF"
+    port_num            = 1*3DIGIT                              ; port number of bridge port
+    port_role           = 1DIGIT                                ; port role 1-root, 2-designated, 3-alternate, 4-backup
+    path_cost           = 1*9DIGIT                              ; port path cost (1 to 200000000)
+    priority            = 3*DIGIT                               ; port priority (0 to 240, DEF:128)
+    port_state          = 1DIGIT                                ; 0-listen, 1-learn, 2-block, 3-forward, 4-disabled
+    desig_cost          = 1*9DIGIT                              ; designated cost
+    desig_root          = 16HEXDIG                              ; designated root
+    desig_bridge        = 16HEXDIG                              ; designated bridge
+    desig_port          = 1*3DIGIT                              ; designated port
+    fwd_transitions     = 1*5DIGIT                              ; number of forward transitions
+    bpdu_sent           = 1*10DIGIT                             ; bpdu transmitted
+    bpdu_received       = 1*10DIGIT                             ; bpdu received
+    tc_sent             = 1*10DIGIT                             ; tc transmitted
+    tc_received         = 1*10DIGIT                             ; tc received
+    bpdu_guard          = "yes" / "no"                          ; port disabled due to bpdu-guard
+
+### STP_PORT_STATE_TABLE
+    ;Stores STP port state per instance
+    ;Status: work in progress
+    key                 = STP_PORT_STATE:ifname:instance        ; ifname and the STP instance
+    state               = 1DIGIT                                ; 0-listen, 1-learn, 2-block, 3-forward, 4-disabled 
+
+
+### STP_VLAN_INSTANCE_TABLE 
```

Comment:
does it make sense to add this as an attribute in the existing vlan table?

Reply:
Yeah this can be moved to existing vlan table. We felt keeping the tables separate will allow us to delete the STP table when the feature is disabled, whereas keeping with vlan table the entry will still be present but needs to be populated with invalid value. Let me know your thoughts, we can change accordingly.

Reply:
Keeping the table separate